### PR TITLE
Clean up F# OSS build targets files.

### PR DESCRIPTION
**Pull request by KevinRansom**
1.  Use public sign.
2.  fix some slightly irritating indentation.
3. Move release and debug speciffic settings to settings.targets
4. Remove TargetFramework support for platforms that are not part of the OSS build.

### CODE REVIEW

**Comment by dsyme** [comment]
What is public signing again?   Does FSharp.Core.dll for .NET Framework still get signed by Microsoft with the same strong name?

**Comment by KevinRansom** [comment]
Public signing is kind of a loader trick.  Delay the sign the dll, and mark the signed bit in the CLR header to signed and the CLR loads the dll fine, except where it would normally verify the assembly,  So public signed assemblies can not be used in Partial App Domains or from the GAC.

Official Microsoft Binaries will still be fully signed using the .net framework keypair.  This is just a way of removing the administrator and skip verification requirement when building our OSS binaries

Kevin

File: src/FSharpSource.Settings.targets [hunk 1]
```diff
@@ -25,11 +25,10 @@
 
   <!-- Standard interpretations of Debug and Release configurations -->
   <PropertyGroup Condition=" '$(Configuration)' == 'Debug' ">
-    <DebugType Condition=" '$(DebugType)' == '' ">full</DebugType>
+    <DebugType Condition=" '$(DebugType)' == '' ">portable</DebugType>
```
Comment (otawfik-ms) [inline src/FSharpSource.Settings.targets]: This will affect all projects?

Comment (KevinRansom) [inline src/FSharpSource.Settings.targets]: VS2014 and Dev15 from Update 2 onwaqrds can both debug portable PDB's, and portable PDBs are much ... much smaller.

**Pull request merged**